modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache

```diff
@@ -73,23 +73,20 @@
       update_params_for_auth! header_params, query_params, opts[:auth_names]
 
       req_opts = {
-        :method => http_method,
-        :headers => header_params,
-        :params => query_params,
         :params_encoding => @config.params_encoding,
         :timeout => @config.timeout,
         :verbose => @config.debugging
       }
 
       if [:post, :patch, :put, :delete].include?(http_method)
         req_body = build_request_body(header_params, form_params, opts[:body])
-        req_opts.update :body => req_body
         if @config.debugging
           @config.logger.debug "HTTP request body param ~BEGIN~\n#{req_body}\n~END~\n"
         end
       end
       request.headers = header_params
       request.body = req_body
+      request.options = OpenStruct.new(req_opts)
```

Comment:
@wing328 @NivathaSV8 I've been attempting to use this feature in a generated ruby SDK, as well as add the ability to directly return the binary media data gotten for a file. It appears to me that this PR has broken the functionality of the code fixed in #7842. While the temp file is created, the file is empty, and directly returning the binary data shows that `@stream` is empty. Prior to this PR, this seemed to work properly, and the culprit appears to be this line (89), as removing it allows the temp file to be created properly and the binary media return I'm working on in #11254 to work as well. I'm still fairly new to ruby, so I may be missing something here, but is there any way this issue could be validated? I've generated the SDK with version 5.3.1 of the CLI that was current as of 1/7/2021. I'm happy to provide more of my findings about this issue to help fix it, but I'd like to make sure I'm not incorrectly using this feature first. Any explanation as to why I might be seeing this issue and what could be done to fix it is greatly appreciated, thanks!